### PR TITLE
Add scan and read QR code

### PR DESCRIPTION
**Pull request by tomriddle7**

### CODE REVIEW

File: Assets/Scripts/GameManager.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+        else if (instance != this)
+            Destroy(gameObject);
+
+        DontDestroyOnLoad(gameObject);
+    }
+
+    public void LoadScene(string sceneName)
+    {
+        //Debug.Log("=========click");
```
Comment (IC-H) [inline Assets/Scripts/GameManager.cs]: Please delete your debug code when it is unnecessary.

File: Assets/Scripts/GameManager.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        SceneManager.LoadScene(sceneName);
+        //instance.StartCoroutine(Load(sceneName));
```
Comment (IC-H) [inline Assets/Scripts/GameManager.cs]: Please delete your debug code when it is unnecessary.

File: Assets/Scripts/GameManager.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        //instance.StartCoroutine(FadeOut(instance.faderObj, instance.faderImg));
```
Comment (IC-H) [inline Assets/Scripts/GameManager.cs]: Please delete your debug code when it is unnecessary.

File: Assets/Scripts/ScanQRCode.cs [hunk 4]
```diff
@@ -72,8 +72,13 @@ private void Scan()
...
 		{
 			resultText = Decode(webCamTexture.GetPixels32 (), webCamTexture.width, webCamTexture.height);
-			Debug.Log (resultText);
-		}
+            if(resultText != null)
+            {
+                GameManager.instance.readJson = resultText;
+                GameManager.instance.LoadScene("ReadQRCode");
+            }
+            //Debug.Log (resultText);
```
Comment (IC-H) [inline Assets/Scripts/ScanQRCode.cs]: Please delete your debug code when it is unnecessary.

File: Assets/Scripts/ScanQRCode.cs [hunk 5]
```diff
@@ -87,12 +92,12 @@ public string Decode(Color32[] colors, int width, int height)
 		return null;
 	}
 
-	void OnGUI()
+	/*void OnGUI()
```
Comment (IC-H) [inline Assets/Scripts/ScanQRCode.cs]: Please delete your code when it is unnecessary not comment out.

File: Assets/Scripts/ScanQRCode.cs [hunk 6]
```diff
@@ -72,8 +72,13 @@ private void Scan()
 		if (webCamTexture != null && webCamTexture.width > 100) 
 		{
 			resultText = Decode(webCamTexture.GetPixels32 (), webCamTexture.width, webCamTexture.height);
-			Debug.Log (resultText);
-		}
+            if(resultText != null)
```
Comment (IC-H) [inline Assets/Scripts/ScanQRCode.cs:69]: Please convert tab to 4 space.
This files dominant indent is tab, but as you know we decided default indent to 4 space

File: Assets/Scripts/GameManager.cs [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    public static GameManager instance = null;
+
+    public string readJson;
+
+    void Awake()
+    {
+        if (instance == null)
+            instance = this;
+
+        else if (instance != this)
```
Comment (IC-H) [inline Assets/Scripts/GameManager.cs:19]: Do you think it can be occur?

File: Assets/Scripts/GameManager.cs [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+            Destroy(gameObject);
```
Comment (IC-H) [inline Assets/Scripts/GameManager.cs:20]: Please explain why you destroy gameObject here not instance?

File: Assets/Scripts/GameManager.cs [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+
+        DontDestroyOnLoad(gameObject);
```
Comment (IC-H) [inline Assets/Scripts/GameManager.cs:22]: Please explain why you DontDestroyOnLoad gameObject here not this?

File: Assets/Scripts/GameManager.cs (hunk 7 above)
Comment (tomriddle7) [reply Assets/Scripts/GameManager.cs:19]: When we change scene, that happens.

File: Assets/Scripts/GameManager.cs (hunk 8 above)
Comment (tomriddle7) [reply Assets/Scripts/GameManager.cs:20]: When we change scene, existing class's instance is this. But new class's instance is not this. So, the gameobject that has new class is destroy.

File: Assets/Scripts/GameManager.cs (hunk 9 above)
Comment (tomriddle7) [reply Assets/Scripts/GameManager.cs:22]: This is to maintain class and inner data when change scene. When change scene, every existing object is destroyed. But if you use 'DontDestroyOnLoad', that object is not destroyed.

**Review by IC-H — APPROVED** [review]

**Pull request merged**